platform-sdk/swirlds-common/src/main/java/com/swirlds/common/io/streams/SerializableDataInputStream.java

```diff
@@ -141,12 +169,18 @@ protected void recordClass(final Object o) {
      * Same as {@link #readSerializable(boolean, Supplier)} except that the constructor takes a class ID
      */
     private <T extends SelfSerializable> T readSerializable(
-            final boolean readClassId, final CheckedFunction<Long, T, IOException> serializableConstructor)
+            final boolean readClassId,
+            @NonNull final CheckedFunction<Long, T, IOException> serializableConstructor,
+            @Nullable final Set<Long> permissibleClassIds)
             throws IOException {
 
         final Long classId;
         if (readClassId) {
             classId = readLong();
+            if (permissibleClassIds != null && !permissibleClassIds.contains(classId)) {
+                throw new IOException("Class ID " + classId + " is not in the set of permissible class IDs");
```

Comment:
Minor: in case the `permissibleClassIds` is expected to have a reasonable size, you might want to include it in the exception message to aid debugging. The `Set.toString()` usually produces a reasonable output unless you use an exotic implementation for this set.

Reply:
added